crates/wasm-encoder/src/linking.rs

```diff
@@ -97,7 +97,7 @@ const WASM_SYMBOL_TABLE: u8 = 8;
 /// A subsection of the [linking custom section][crate::LinkingSection] that
 /// provides extra information about the symbols present in this Wasm object
 /// file.
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Default)]
```

Comment:
With `Default` being added could the existing `new` functions defer to `Default`?

Reply:
(ideally done for all the other `Default` additions in this PR as well)

Reply:
I'll update all places that now derive `Default` and change their new impls.